e2e/integration/wordpress-01/yoast-package.spec.js

```diff
@@ -13,14 +13,35 @@ describe("Yoast Package", () => {
     cy.task("removeAllPlugins");
   });
 
+  /**
+   * Visit the specified link with a request and returned the parsed, static, dom.
+   *
+   * @param link - The link of the page.
+   * @returns Returns the parsed response from the request.
+   */
+  const visitLinkAndParseDom = async (link) => {
+    const req = await cy.request(
+      `http://localhost:3001${link}?frontity_name=yoast-package`
+    );
+
+    const parser = new DOMParser();
+
+    return parser.parseFromString(req.body, "text/html");
+  };
+
   /**
    * Check the title for the current page is the given one.
    *
+   * @param link - The given link to execute the request.
    * @param title - The page title for the given link.
    */
-  const checkTitle = (title) => {
-    it("should render the correct title", () => {
-      cy.get("title").should("contain", title);
+  const checkTitle = (link, title) => {
+    /* eslint-disable-next-line cypress/no-async-tests */
+    it("should render the correct title", async () => {
+      const dom = await visitLinkAndParseDom(link);
+      const el = dom.querySelector("title");
+
+      cy.get(el).should("contain", title);
```

Comment:
If I change the value of `title` here to be a random string the test passes as well, so the `async` function is not working as expected.

By the way, I tried to follow the Cypress way and, rewriting the `checkTitle` function like this:

```ts
const checkTitle = (link, title) => {
  it("should render the correct title", () => {
    cy.request(`http://localhost:3001${link}?frontity_name=yoast-package`)
      .then((response) => parseHTML(response.body))
      .then((html) => {
        cy.get(html.querySelector("title")).should("contain", title);
      });
  });
};
```
then I get the following error:

![Screenshot from 2021-01-20 13-34-25](https://user-images.githubusercontent.com/6917969/105176118-15a53c00-5b25-11eb-8fd9-6c0cdeda77dc.png)

Reply:
wow! Such a great suggestion @DAreRodz! 🙇‍♂️  Will implement it and remove the `async` bits.